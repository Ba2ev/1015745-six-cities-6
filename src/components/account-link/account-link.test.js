import React from "react";
import {render} from '@testing-library/react';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import AccountLink from './account-link';


const user = {
  "email": `test@test.ru`
};

const mockStore = configureStore({});

jest.spyOn(redux, `useDispatch`);

it(`AccountLink should render correctly`, () => {
  const history = createMemoryHistory();

  const {getByText} = render(
      <redux.Provider store={mockStore({})}>
        <Router history={history}>
          <AccountLink account={user.email}/>
        </Router>
      </redux.Provider>
  );

  const linkSignOut = getByText(`Sign out`);
  const textEmail = getByText(user.email);

  expect(linkSignOut).toBeInTheDocument();
  expect(textEmail).toBeInTheDocument();
});
