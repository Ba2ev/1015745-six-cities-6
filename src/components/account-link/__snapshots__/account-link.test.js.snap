// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountLink should render snapshot correctly 1`] = `
<div>
  <a
    class="header__nav-link header__nav-link--profile"
    href="/favorites"
  >
    <div
      class="header__avatar-wrapper user__avatar-wrapper"
    />
    <span
      class="header__user-name user__name"
    >
      test@test.ru
    </span>
  </a>
  <a>
    Sign out
  </a>
</div>
`;
